src/AutoRest.CSharp/Common/Generation/Writers/CodeWriterExtensions.cs

```diff
@@ -31,6 +42,43 @@ public static CodeWriter AppendNullableValue(this CodeWriter writer, CSharpType
             return writer;
         }
 
+        public static CodeWriter.CodeWriterScope WriteMethodDeclaration(this CodeWriter writer, MethodSignature method)
+        {
+            writer.WriteXmlDocumentationSummary(method.Description);
+            writer.WriteXmlDocumentationParameters(method.Parameters);
+            writer.WriteXmlDocumentationRequiredParametersException(method.Parameters);
+            if (method.ReturnDescription != null)
+            {
+                writer.WriteXmlDocumentationReturns(method.ReturnDescription);
+            }
+
+            writer
+                .Append($"{method.Modifiers} ")
+                .AppendIf($"{method.ReturnType} ", method.ReturnType != null)
```

Comment:
Don't we need to write `void` here?

Reply:
This needs further work. Cause right now there is no way to differentiate void method from constructor or finalizer. Probably, there should be another property (I don't think we need a separate type like `MethodInfo`/`ConstructorInfo`).

Reply:
But should be enough for now.